index.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+class WebError extends Error {
+  constructor(status, message) {
+    super(`${status}: ${message}`);
+    this.status = status;
+    return this;
+  }
+}
+
+function minuscule(app) {
+
+  const prodLike = process.env.ENV === 'production';
+
+  const self = {
+
+    use(fn) {
+      app.use(async (req, res, next) => {
+        try {
+          await fn(req);
+          return next();
+        } catch (e) {
+          return self.handleError(req, e);
+        }
+      });
+    },
+
+    route(method, path, ...fns) {
+      if (((typeof method) !== 'string') || ((typeof path) !== 'string') || ((typeof fns[0]) !== 'function')) {
+        throw new Error(`route() must be called with (method, path, [...optionalMiddlewareFns], fn)`);
+      }
+      app[method](path, async req => {
+        let result;
+        for (const fn of fns) {
+          try {
+            result = await fn(req);
+          } catch (e) {
+            return self.handleError(req, e);
+          }
+        }
+        return req.res.send(result);
+      });
```

Comment:
If `fns` is defined as an array of middleware, we can write the following instead

```suggestion
      app[method](path, ...fns);
```

Reply:
Unfortunately we cannot because even in Express 5 there is no support for continuing the chain on a *successful* promise, and also no support for continuing the chain on a successful synchronous return for that matter. Which is important because minuscule middleware continues the chain by default — because that is usually desirable. Because minuscule is focused on APIs, it doesn't currently have any support for middleware "taking over" and delivering the response on its own, but if that actually comes up, I would implement it by returning a special value because it is an edge case and I want to keep the more typical pathways (errors and success) as simple as possible.